NEXTCLOUD.cmake

```diff
@@ -15,6 +15,34 @@ set( APPLICATION_VIRTUALFILE_SUFFIX "nextcloud" CACHE STRING "Virtual file suffi
 set( LINUX_PACKAGE_SHORTNAME "nextcloud" )
 set( LINUX_APPLICATION_ID "${APPLICATION_REV_DOMAIN}.${LINUX_PACKAGE_SHORTNAME}")
 
+option(DEV_BUILD "If enabled, name variables will be appended with DEV_SUFFIX in order to reduce conflicts." OFF)
+if (NOT DEV_SUFFIX)
+  set(DEV_SUFFIX "Dev"
+    CACHE STRING
+      "Choose a suffix with which to append build name variables to prevent conflicts."
+  FORCE)
+endif ()
+
+function(ADD_SUFFIXES SUFFIX_STRING)
+    string(TOLOWER ${SUFFIX_STRING} SUFFIX_LOWER)
+    string(APPEND APPLICATION_NAME " (${SUFFIX_STRING} Build)")
+    set(APPLICATION_NAME ${APPLICATION_NAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_SHORTNAME "${SUFFIX_STRING}")
+    set(APPLICATION_SHORTNAME ${APPLICATION_SHORTNAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_EXECUTABLE "${SUFFIX_LOWER}")
+    set(APPLICATION_EXECUTABLE ${APPLICATION_EXECUTABLE} PARENT_SCOPE)
+    string(APPEND APPLICATION_REV_DOMAIN "${SUFFIX_LOWER}")
+    set(APPLICATION_REV_DOMAIN ${APPLICATION_REV_DOMAIN} PARENT_SCOPE)
+    string(APPEND LINUX_PACKAGE_SHORTNAME "${SUFFIX_LOWER}")
+    set(LINUX_PACKAGE_SHORTNAME ${LINUX_PACKAGE_SHORTNAME} PARENT_SCOPE)
+    string(APPEND LINUX_APPLICATION_ID "${SUFFIX_LOWER}")
+    set(LINUX_APPLICATION_ID ${LINUX_APPLICATION_ID} PARENT_SCOPE)
+endfunction()
```

Comment:
I would not put that into a function. It is called in only in one place and I doubt that this will change in the future. Also this statements that set variables in the parent scope introduce a lot of noise. If you want you can use a macro or just inline it in the `if` statement. In general I try to omit functions that have a lot of side effects and modify the parent scope because that always can introduce some danger if the caller is not aware of it.

Reply:
IIRC I only made it a function because you asked me to, but I can change it back.